Client/Frontend/Menu/Items/MenuItemView.swift

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+/**
+ * creates a view that consists of an image and a title.
+ * the image view is displayed at the center top of the
+ **/
+public class MenuItemView: UIControl {
+
+    var padding: CGFloat = 5.0
+
+    private(set) public var imageView: UIImageView
+    private(set) public var titleLabel: UILabel
+
+    private var previousLocation: CGPoint?
+    public init() {
+        imageView = UIImageView()
+        titleLabel = UILabel()
+
+        super.init(frame: CGRectZero)
+
+        titleLabel.lineBreakMode = NSLineBreakMode.ByWordWrapping
+        titleLabel.numberOfLines = 2
+        titleLabel.textAlignment = NSTextAlignment.Center
+
+        self.addSubview(imageView)
+
+        self.addSubview(titleLabel)
+
+        imageView.snp_makeConstraints { make in
+            make.centerX.equalTo(self)
+            make.bottom.equalTo(self.snp_centerYWithinMargins)
+        }
+
+        titleLabel.snp_makeConstraints { make in
+            make.top.equalTo(imageView.snp_bottom).offset(padding)
+            make.centerX.equalTo(self)
+            make.bottom.equalTo(self).offset(-padding)
+            make.width.lessThanOrEqualTo(self.snp_width).dividedBy(1.5)
+        }
```

Comment:
Would it make it easier to setup the constraints so that:

1. The imageView is pinned to the top by a margin, centered X
2. Pin the top of the label to the bottom of the image and left/right using a margin then pinning the bottom to a greaterThanOrEqual to bottom self with a margin.

If you set the number of lines to 0 and have it flow, the constraints should hold correctly regardless of 1, 2, or 3 lines. Would also clean up the pinning to the width and division by 1.5.